nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -35,49 +42,117 @@ public NsisExecutable() {
 	 * Creates and initializes a Nsis Executable object
 	 * 
 	 * @param factory      that will be used to create Nsis Executable
-	 * @param byteProvider object that will permit reading bytes from the file.
-	 *                     The lifespan of the byte provider is controlled by
-	 *                     Ghidra.
+	 * @param byteProvider object that will permit reading bytes from the file. The
+	 *                     lifespan of the byte provider is controlled by Ghidra.
 	 * @param layout       object used to load PE executables
 	 * @return The Nsis executable object
 	 * @throws IOException
 	 * @throws InvalidFormatException
 	 */
-	public static NsisExecutable createNsisExecutable(GenericFactory factory,
-			ByteProvider bp, SectionLayout layout)
-			throws IOException, InvalidFormatException {
-		NsisExecutable nsisExecutable = (NsisExecutable) factory
-				.create(NsisExecutable.class);
+	public static NsisExecutable createNsisExecutable(GenericFactory factory, ByteProvider bp,
+			SectionLayout layout) throws IOException, InvalidFormatException {
+		NsisExecutable nsisExecutable = (NsisExecutable) factory.create(NsisExecutable.class);
 		nsisExecutable.initNsisExecutable(factory, bp, layout);
 		return nsisExecutable;
 	}
 
-	private void initNsisExecutable(GenericFactory factory, ByteProvider bp,
-			SectionLayout layout) throws IOException, InvalidFormatException {
-		this.reader = new FactoryBundledWithBinaryReader(factory, bp,
-				/* isLittleEndian= */ true);
+	private void initNsisExecutable(GenericFactory factory, ByteProvider bp, SectionLayout layout)
+			throws IOException, InvalidFormatException {
+		this.reader = new FactoryBundledWithBinaryReader(factory, bp, /* isLittleEndian= */ true);
 		this.headerOffset = findHeaderOffset();
-		initScriptHeader(bp);
+		initScriptHeader();
+		if ((this.scriptHeader.compressedHeaderSize & 0x80000000) != 0) { // Check
```

Comment:
Please make this value a named constant, for example FLAG_COMPRESSION_USED

Reply:
Even if it is a bitmask?